app/src/main/java/com/deitel/tipcalculator/MainActivity.java

```diff
@@ -112,6 +125,33 @@ public void afterTextChanged(Editable s) { }
       public void beforeTextChanged(
          CharSequence s, int start, int count, int after) { }
    };
+
+   private final TextWatcher PartyWatcher = new TextWatcher() {
+      // called when the user modifies the bill amount
+      @Override
+      public void onTextChanged(CharSequence s, int start,
+                                int before, int count) {
+
+         try { // get bill amount and display currency formatted value
+            people = Double.parseDouble(s.toString());
+
+
+         }
+         catch (NumberFormatException e) { // if s is empty or non-numeric
+
```

Comment:
Include at least a log error here so that you can troubleshoot errors. My guess is that the catch may be triggered on your repeat attempts for some reason.